File: pigatize.rb
def pigatize(text)
  if starts_with_vowel?(text)
    text = text + "way"
  else
    consonant = text[0]
    text[1..-1] + consonant + "ay"
  end
end
 
def starts_with_vowel?(text)
  arr_vowel = ["a","e","i","o","u"]
  arr_vowel.include?(text[0])
end
 
loop do
	puts "Please enter a word and I will translate to pig latin"
	text = gets.chomp.downcase
	break if text.length == 0 # Break out of the loop if I say nothing
	puts pigatize(text)
end

File: pigatize.js
var readline = require("readline");

var starts_with_vowel = function(text) {
  return ( text[0].search(/[a|e|i|o|u]/i) >= 0 );
}

var pigatize = function(text) {
  if ( starts_with_vowel(text) ) {
    return text + "way";
  } else {
    return text.slice(1) + text[0] + "ay";
  }
}

var rl = readline.createInterface(process.stdin, process.stdout);
rl.setPrompt("Please enter a word and I will translate to pig latin: ");
rl.prompt();

rl.on('line', function(line) {
  if (line.length === 0) {
    process.exit(0);
  }
  console.log(pigatize(line));
  rl.prompt();
}).on('close', function() {
  process.exit(0);
});
